app/presenters/openid_connect_user_info_presenter.rb

```diff
@@ -8,29 +8,55 @@ def initialize(identity)
   end
 
   def user_info
-    user = identity.user
-
-    {
+    info = {
       sub: identity.uuid,
       iss: root_url,
-      email: user.email,
+      email: identity.user.email,
       email_verified: true
     }.merge(loa3_attributes)
+
+    OpenidConnectAttributeScoper.new(identity.scope).filter(info)
   end
 
   private
 
   def loa3_attributes
+    phone = loa3_data.phone
+
     {
       given_name: loa3_data.first_name,
       family_name: loa3_data.last_name,
       middle_name: loa3_data.middle_name,
       birthdate: loa3_data.dob,
+      address: address,
+      phone: phone,
+      phone_verified: phone ? true : nil
+    }
+  end
+
+  def address
+    return nil if loa3_data.address1.blank?
```

Comment:
Perhaps clearer to be explicit, but no need to return `nil` here (tested in console just to make sure)

```
irb(main):015:0> def thing
irb(main):016:1>    return nil if true == true
irb(main):017:1> end
=> :thing
irb(main):018:0> thing
=> nil
irb(main):019:0>  def thing
irb(main):020:1> return if true == true
irb(main):021:1> end
=> :thing
irb(main):022:0> thing
=> nil
```